### PR TITLE
[Icon] Add cancel major

### LINKED ISSUE #7490: Add Cancel major icon and deprecate mobile icon version
## Feature request summary

- [ ] Add `CancelMajor` icon version for `Cancel small` equivalent
- [ ] Rename `Cancel small` to `CancelMinor`

## Rationale

- Inconsistency in major/minor versions of the Cancel icon, as well as a legacy mobile-only version

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s :+1: upvotes to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md) for more information.

### PR DESCRIPTION
**Pull request by anthonymenecola**
### WHY are these changes introduced?

Add Major icon version for `Cancel` to accompany minor version.
Duplicated minor `Cancel small` to get naming for major/minor on the same track. 

Deprecating `MobileCancelMajor`, and `Cancel small` in a separate PR.

### WHAT is this pull request doing?

Added major version for `Cancel` icon. Solves part of this issue https://github.com/Shopify/polaris/issues/7490 

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

**Comment by anthonymenecola** [comment]
@alex-page @sarahill did you want me to include the changes to rename `Cancel small` to `Cancel` since it is specified within the Minor set? What about the deprecation of `Mobile cancel major`? Let me know and I can add to this PR. Thanks

**Comment by alex-page** [comment]
@anthonymenecola in this PR can you add a copy of the `Cancel small` icon and rename it to `Cancel`. Then we can deprecate `Cancel small` and `MobileCancelMajor` in a follow up PR.

File: .changeset/cyan-jars-punch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-icons': major
```
Comment (alex-page) [inline .changeset/cyan-jars-punch.md]:
```suggestion
'@shopify/polaris-icons': minor
```

The major change would be when we remove the old icons

**Review by alex-page — APPROVED** [review]

**Review by sarahill — APPROVED** [review]
🎉

**Pull request merged**